photonjs/README.md

```diff
@@ -208,6 +214,119 @@ const deletedCount: number = await photon.user.delete().count()
 uts
 ```
 
+## Field-level Primary Key constraint
+
+Different unique constraints will change Photon's `where` blocks:
+
+````groovy
+model User {
+    id String @id
+    firstName String
+    lastName String
+    email String
+}
+
+```ts
+photon.user.find({ id: 10 })
+````
+
+### Field-level unique constraint
+
+```groovy
+model User {
+    id String @id
+    firstName String
+    lastName String
+    email String  @unique
+}
+```
+
+```ts
+// allowed
+photon.user.find({ id: 10 })
+photon.user.find({ email: "alice@prisma.io" })
+// compiler error
+photon.user.find({ id: 10, email: "alice@prisma.io" })
+```
+
+### Model-level composite constraint (unnamed)
+
+```groovy
+model User {
+    id String @id
+    firstName String
+    lastName String
+    email String @unique
+    @@unique([ firstName, lastName ])
+}
+```
+
+```ts
+// allowed
+photon.user.find({ id: 10 })
+photon.user.find({ email: "alice@prisma.io" })
+photon.user.find({
+  firstName_lastName: {
+    firstName: "Alice",
+    lastName: "Prisma"
+  }
+})
+// compiler error
+photon.user.find({
+  email: "alice@prisma.io",
+  firstName_lastName: {
+    firstName: "Alice",
+    lastName: "Prisma"
+  }
+})
+photon.user.find({
+  id: 10,
+  firstName_lastName: {
+    firstName: "Alice",
+    lastName: "Prisma"
+  }
+})
+```
+
+### Naming the composite constraint
+
+```groovy
+model User {
+    id String @id
+    firstName String
+    lastName String
+    email String @unique
+    @@unique([ firstName, lastName ], name: "fullName")
```

Comment:
We have already implemented the `name` argument but with different semantics. Currently the `name` argument refers to the name of the underlying unique index in the database. In your example it insteads influences the name of the combined criterion in the Photon API. I think we should distinguish those two names as they are different concepts, e.g. `dbName` vs `photonName` (not good i know. just to illustrate my point)
The reason we need different names is that the name of the underlying index must be unique in some databases e.g. Postgres.  And i think we do want to support configuring the index name.

@matthewmueller : I just discovered this by interpreting your desired semantics. We should clearly document what the arguments for each attribute mean like in the datamodel v1.1 [spec](https://github.com/prisma/prisma/issues/3405).

Reply:
Ah okay that makes sense! I just realized it's also in here: https://github.com/prisma/specs/tree/master/schema#core-block-attributes but that could use some clarification.

Reply:
I agree that we should distinguish, I'll think about some options there 👍

Reply:
Just discovered that this is the photon spec here. So my last sentence does not make sense here as it applies to the PSL spec.

Reply:
I've renamed this to `alias`.